packages/cli/src/WorkflowRunner.ts

```diff
@@ -187,42 +188,44 @@ export class WorkflowRunner {
 			);
 		} else if (executionsProcess === 'main') {
 			executionId = await this.runMainProcess(data, loadStaticData, executionId, responsePromise);
+			void Container.get(InternalHooks).onWorkflowBeforeExecute(executionId, data);
 		} else {
 			executionId = await this.runSubprocess(data, loadStaticData, executionId, responsePromise);
+			void Container.get(InternalHooks).onWorkflowBeforeExecute(executionId, data);
 		}
 
-		void Container.get(InternalHooks).onWorkflowBeforeExecute(executionId, data);
-
-		const postExecutePromise = this.activeExecutions.getPostExecutePromise(executionId);
-
-		const externalHooks = Container.get(ExternalHooks);
-		postExecutePromise
-			.then(async (executionData) => {
-				void Container.get(InternalHooks).onWorkflowPostExecute(
-					executionId!,
-					data.workflowData,
-					executionData,
-					data.userId,
-				);
-			})
-			.catch((error) => {
-				ErrorReporter.error(error);
-				console.error('There was a problem running internal hook "onWorkflowPostExecute"', error);
-			});
-
-		if (externalHooks.exists('workflow.postExecute')) {
+		// these calls are moved into the worker
+		if (executionsMode !== 'queue' || data.executionMode === 'manual') {
+			const postExecutePromise = this.activeExecutions.getPostExecutePromise(executionId);
+			const externalHooks = Container.get(ExternalHooks);
 			postExecutePromise
```

Comment:
This could also totally be another hook, there's no reason for this code to be here right? This is just yet another telemetry event. No need to change this now, but we have to do this cleanup. Maybe add a comment so we tackle this in the next iterations.

Reply:
refactored